lib/coverband/utils/source_file.rb

```diff
@@ -81,9 +81,27 @@ def status
       # The array of coverage data received from the Coverage.result
       attr_reader :coverage
 
-      def initialize(filename, coverage)
+      # the date this version of the file first started to record coverage
+      attr_reader :first_updated_at
+      # the date this version of the file last saw any coverage activity
+      attr_reader :last_updated_at
+      NOT_AVAILABLE = 'not available'
+
+      def initialize(filename, file_data)
         @filename = filename
-        @coverage = coverage
+        if file_data.is_a?(Hash)
```

Comment:
Is this needed for backwards compatibility?

Reply:
This is needed for files that aren't in the coverage data and need to get reported at `0%`

Reply:
Oh gotcha. I guess we could eventually simplify this by just having everything default to a hash?

Reply:
yeah, I think if I do some further refactoring we can change remove these checks... and we can remove that other line I added a todo to remove... but it will required a bit more of a rewrite than I want to do for fast-tracking it into the release ;)

Reply:
👍